src/language-js/printer-estree.js

```diff
@@ -2886,7 +2898,8 @@ function printPathNoParens(path, options, print, args) {
 
       return group(concat(parts));
     }
-    case "TypeCastExpression": {
+    case "TypeCastExpression":
+    case "TSTypeCastExpression": {
```

Comment:
This `case` isn't needed. The AST shouldn't contain such nodes.

Reply:
226b1db :+1: